src/redux/message-service/saga.ts

```diff
@@ -100,6 +106,8 @@ function* signMessage(data) {
   while (signatureResponse.requestId !== requestId) {
     signatureResponse = yield take(actionFilter);
   }
+
+  yield put(walletActions.storeMessageRequest(data, signatureResponse.signature, Direction.Sent));
```

Comment:
Same comment as above: this should probably be done inside the wallet as part of the signature request.

Reply:
I think the reason it was originally split out was due to the fact that we want the wallet to be able to sign more than states (Like the address value that is sent for `withdraw`

Reply:
Wouldn't signing the final state always happen within the context of the wallet though? E.g. I open my channel wallet and instruct it to send the right transaction to metamask to withdraw to a given address (which will involve signing it). I might have missed other use cases though. What do you think?